pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -596,7 +596,7 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 
 			err := s.pushConnection(con, pushEv)
 			if err != nil {
-				return nil
+				return err
```

Comment:
@costinm was there a reason we were returning nil previously?

Reply:
The code changed here: https://github.com/istio/istio/commit/d455f807ab5191073a70e0d2973b9280869bf6f7#diff-14f4a8a239bf5ba82053cf6606daace5R496

Maybe just a slip of pen?

It was originally returning error.

Reply:
ping @costinm